packages/expo-battery/ios/BatteryModule.swift

```diff
@@ -0,0 +1,113 @@
+import ExpoModulesCore
+
+let batteryLevelDidChange: String = "Expo.batteryLevelDidChange"
+let batteryStateDidChange: String = "Expo.batteryStateDidChange"
+let powerModeDidChange: String = "Expo.powerModeDidChange"
+
+public class BatteryModule: Module {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoBattery")
+
+    Constants([
+      "isSupported": isSupported()
+    ])
+
+    Events(batteryLevelDidChange, batteryStateDidChange, powerModeDidChange)
+
+    AsyncFunction("getBatteryLevelAsync") { () -> Float in
+      return UIDevice.current.batteryLevel
+    }
+
+    AsyncFunction("isLowPowerModeEnabledAsync") { () -> Bool in
+      return ProcessInfo.processInfo.isLowPowerModeEnabled
+    }
+
+    AsyncFunction("getBatteryStateAsync") { () -> Int in
+      switch UIDevice.current.batteryState {
```

Comment:
Looks like `UIDevice.BatteryState` has the same cases and values as ours `BatteryState`, so maybe we can just return `UIDevice.current.batteryState.rawValue` here? 🤔 If so, then please also add a comment that this type is in-sync with our TS enum.

Reply:
If that's the case, then yeah - we can definitely do that. I agree on the comment remark, we should definitely include so we're aware of what we do if the enums ever become "out-of-sync"